framework/include/convergence/Convergence.h

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseObject.h"
+#include "SetupInterface.h"
+#include "PerfGraphInterface.h"
+
+class Convergence : public MooseObject, public SetupInterface, public PerfGraphInterface
+{
+public:
+  static InputParameters validParams();
+
+  enum class MooseAlgebraicConvergence
```

Comment:
I suggest `MooseConvergenceStatus` (I guess we need the "Moose" part to distinguish from PETSc's convergence return type?)

Reply:
Yes But I kinda wanted to indicate this is purely Algebraic convergence, there might be other types like Spatial, Temporal. I could add status if you think it's relevant, but maybe it gets too long.